libsolidity/lsp/FileRepository.cpp

```diff
@@ -51,6 +67,8 @@ string FileRepository::sourceUnitNameToUri(string const& _sourceUnitName) const
 		return "file:///" + _sourceUnitName;
```

Comment:
Regarding the first part of the `if` above (I can't comment on it directly):
```c++
if (m_sourceUnitNamesToUri.count(_sourceUnitName))
		return m_sourceUnitNamesToUri.at(_sourceUnitName);
```
Is the assumption that this is already an URI, i.e. starts with `protocol://`? If so then maybe we should have an assert here?

Reply:
Yes, right below, the function named `setSourceByUri`  This is called in a context where the remote end is providing us URIs. I'm not sure an assert() is really necessary (?) as in that case, the client would just receive as URI what it initially has sent as URI (if it could be mapped to a local file). Hmmm.

Reply:
The thing I'm not sure about is: can a non-URI end up in `m_sourceUnitNamesToUri`? My impression from this function is that it's trying hard to always have the result prefixed with `file://` but it wasn't obvious to me if this particular `if` always guarantees that prefix too. My guess is yes and if that's the case, an assert would make it obvious. If not, I think it needs a comment saying how/why a non-URL can end up in `m_sourceUnitNamesToUri`.